src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanManager.java

```diff
@@ -61,10 +61,34 @@ public Plan getPlan() {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(List<Block> dirtiedAssets) {
```

Comment:
Glad that you brought this up. I wanted to also use Collection, because then we can also pass in Set. 

Another thing that I want to discuss while we are at it is, we need to think a bit more on this interface.

`Collection<Block>` vs `Collection<String>`. Because, current we assume that two blocks trying to act on the same task should be named same. This is currently not enforced, and can lead to bugs.

Moving `Collection<String>` is slightly better, because then we are stating that it contains the name of the assets being worked upon.

Although, I'm also not fully convinced yet with above. Thoughts?